float64 traj
float64 dtraj
float64 d2traj
float64 d3traj
float64 t
